### PR TITLE
[GPU] Clamp f16 INF inputs in RMS GPU kernels to prevent NaN output.

### PR DESCRIPTION
**Pull request by zhanmyz**
### Details:
  - Clamp f16 INF inputs in RMS GPU kernels to prevent NaN output.
### Description of the issue (symptom, root-cause, how it was resolved)
  - **Symptom**:
    - ParlerTTS Mini text_encoder (T5-based, `google/flan-t5-base`) produced all-NaN outputs on GPU FP16.
    - GPU FP32 and CPU modes passed.

  - **Root Cause**:
    - The first INF appears at:
        - `__module.encoder.block.1.layer.1.DenseReluDense.wo/aten::linear/MatMul`
        - prefix dump: `INF_COUNT=5`, `FINITE_MAX=60928`
    - The T5 FFN down-projection MatMul (`block.1.layer.1.DenseReluDense.wo`) produces 5 output elements exceeding FP16 max (65504). When stored in the f16 output buffer, they become INF. These INF values propagate into the downstream RMS kernel, which computes `rsqrt(mean(INF²) + eps) = rsqrt(INF) = 0`, then `INF × 0 = NaN` (IEEE 754 indeterminate form). The 5 NaN values then cascade through the attention mechanism: NaN in Q/K rows → NaN in attention scores columns → softmax(NaN)=NaN for all rows → 100% NaN attention output → all downstream layers produce NaN.

  - **Resolution**:
    - In `rms_gpu_bfyx_opt.cl` and `rms_gpu_ref.cl`, clamp f16 input values to ±`FP16_MAX_FINITE` (65504) before squaring in the RMS accumulation.
    - Conditional on `INPUT0_TYPE_SIZE == 2` (f16 only); f32 path is a no-op macro `#define RMS_CLAMP(val) (val)`.
    - For any finite f16 value: `clamp(v, -65504, 65504) = v` (mathematical identity, zero impact on any model).
    - Only INF values are affected: `clamp(INF) = 65504` → finite accumulation → non-zero rsqrt → no NaN.

#### Reproduction step and snapshot (if applicable. Do not attach for customer model)
- python -m pytest test.py \
    --tb=native \
    --env_conf=.automation/env_config.yml \
    --test_conf=.automation/test_configs/desktop_test_config_gpu_llm.yml \
    -m "not launch_only_if_manually_specified" \
    --pregen_irs=irs_mapping.csv \
    --tf_models_version=1.15.2 \
    --modules pipelines/production/pytorch/heavy/parler_tts.py \
    -k "Pytorch_ParlerTTSMini" \
    --dynamism_type=None \
    --log-cli-level INFO

#### Problematic graph
image

#### Checklist
 - [x] Is it a proper fix? (not a workaround)
 - [x] Did you include test case for this fix, if necessary?
 - [x] Did you review existing test that can be extended to cover this scenario? Which test did you review?

### Tickets:
  - *CVS-187992*

### CODE REVIEW

**Comment by CuriousPanCake** [comment]
I know there're 'do not review' tags on this PR, but isn't it the same thing that was reverted? Are you planning on trying a different solution?

**Comment by zhanmyz** [comment]
> I know there're 'do not review' tags on this PR, but isn't it the same thing that was reverted? Are you planning on trying a different solution?

Yes, I will try a different solution. I submitted this PR primarily to request code review feedback from Copilot.
I added an extra conditional check: `if (m->get_variables().empty())`.
After reviewing the feedback from GitHub Copilot and gaining a deeper understanding of this issue, I realized this change is not a proper fix targeting the root cause. I will keep exploring alternative solutions.
Thanks!

**Comment by wilson-seok** [comment]
If you updated code per copilot review, check resolve it to recognize.
If you changed code and open copilot review is not valid anymore, please run copilot review again.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl [hunk 1]
```diff
@@ -123,7 +131,7 @@ KERNEL(rms_gpu_bfyx_opt)(
         for (; i + 8 <= items_num; i += 8) {
             MAKE_VECTOR_TYPE(INPUT0_TYPE, 8) v = DT_INPUT_BLOCK_READ8(input, ibase + i * sgs);
             unroll_for (int j = 0; j < 8; j++) {
-                ACCUMULATOR_TYPE tmp = TO_ACCUMULATOR_TYPE(v[j]);
+                ACCUMULATOR_TYPE tmp = RMS_CLAMP(TO_ACCUMULATOR_TYPE(v[j]));
```
[the remark this answers is not in the document]
Comment (zhanmyz) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl]: Updated.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl [hunk 2]
```diff
@@ -24,6 +24,14 @@
 #define OUTPUT_VEC_TYPE MAKE_VECTOR_TYPE(OUTPUT_TYPE, SUBGROUP_BLOCK_SIZE)
 #endif
 
+// Clamp f16 INF to prevent NaN in RMS (INF * rsqrt(INF) = INF*0 = NaN).
+#if INPUT0_TYPE_SIZE == 2
```
[the remark this answers is not in the document]
Comment (zhanmyz) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl]: Good catch — fixed. INPUT0_TYPE_SIZE == 2 matched both F16 and BF16 (same storage width, very different dynamic range), so legitimate large BF16 values (e.g. ~1e5–2e5) were being incorrectly clamped to the F16 max, corrupting normalization. The guard is now #if INPUT0_TYPE_SIZE == 2 && !INPUT0_IS_BF16, restricting RMS_CLAMP to genuine F16 inputs only. INPUT0_IS_BF16 is an existing JIT flag (set by jitter.cpp, already used elsewhere in the kernel selector), so no new plumbing was needed. Verified with a new rms_test_bf16_large_values_not_clamped unit test

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl [hunk 3]
```diff
@@ -24,6 +24,14 @@
...
+#define FP16_MAX_FINITE 65504.0f
+#define RMS_CLAMP(val) clamp((val), (ACCUMULATOR_TYPE)(-FP16_MAX_FINITE), (ACCUMULATOR_TYPE)(FP16_MAX_FINITE))
```
[the remark this answers is not in the document]
Comment (zhanmyz) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl]: That means RMS_CLAMP was silently masking genuine upstream NaN bugs instead of surfacing them. Changed the macro to only saturate ±Inf and leave NaN untouched: (isinf(val) ? clamp(val, -FP16_MAX_FINITE, FP16_MAX_FINITE) : (val)). Verified with the consolidated rms_f16_inf_and_nan_input test (batch 0 = ±Inf → clamped as before; batch 1 = quiet NaN → now asserted to remain NaN via std::isnan).

**Comment by zhanmyz** [comment]
This ticket issue can be resolved by using `activations_scale_factor`.
So, close it.
